### ISSUE TITLE
[macOS/BSD] Clang fails to compile

### ISSUE THREAD

**Issue by qyot27** [AviSynth/AviSynthPlus#134]
Using Clang-9 or even the OS default of Clang 8, the following occurs on FreeBSD:

```
[qyot27@bsdhost ~/AviSynthPlus/avisynth-build]$ ninja
[2/6] Building CXX object avs_core/CMakeFiles/AvsCore.dir/core/MTGuard.cpp.o
FAILED: avs_core/CMakeFiles/AvsCore.dir/core/MTGuard.cpp.o 
/usr/local/bin/clang++  -DAvsCore_EXPORTS -DBUILDING_AVSCORE -I../avs_core/include -Iavs_core -msse2  -fPIC   -std=c++17 -MD -MT avs_core/CMakeFiles/AvsCore.dir/core/MTGuard.cpp.o -MF avs_core/CMakeFiles/AvsCore.dir/core/MTGuard.cpp.o.d -o avs_core/CMakeFiles/AvsCore.dir/core/MTGuard.cpp.o -c ../avs_core/core/MTGuard.cpp
../avs_core/core/MTGuard.cpp:65:3: warning: deleting pointer to incomplete type 'std::mutex' may cause undefined behavior [-Wdelete-incomplete]
  delete FilterMutex;
  ^      ~~~~~~~~~~~
../avs_core/core/MTGuard.h:12:9: note: forward declaration of 'std::mutex'
  class mutex;
        ^
../avs_core/core/MTGuard.cpp:68:6: error: reference to 'mutex' is ambiguous
std::mutex* MTGuard::GetMutex() const
     ^
../avs_core/core/MTGuard.h:12:9: note: candidate found by name lookup is 'std::mutex'
  class mutex;
        ^
/usr/include/c++/v1/__mutex_base:40:78: note: candidate found by name lookup is 'std::__1::mutex'
class _LIBCPP_TYPE_VIS _LIBCPP_THREAD_SAFETY_ANNOTATION(capability("mutex")) mutex
                                                                             ^
../avs_core/core/MTGuard.cpp:99:38: error: reference to 'mutex' is ambiguous
        this->FilterMutex = new std::mutex();
                                     ^
../avs_core/core/MTGuard.h:12:9: note: candidate found by name lookup is 'std::mutex'
  class mutex;
        ^
/usr/include/c++/v1/__mutex_base:40:78: note: candidate found by name lookup is 'std::__1::mutex'
class _LIBCPP_TYPE_VIS _LIBCPP_THREAD_SAFETY_ANNOTATION(capability("mutex")) mutex
                                                                             ^
../avs_core/core/MTGuard.cpp:99:33: error: allocation of incomplete type 'std::mutex'
        this->FilterMutex = new std::mutex();
                                ^~~~~~~~~~
../avs_core/core/MTGuard.h:12:9: note: forward declaration of 'std::mutex'
  class mutex;
        ^
../avs_core/core/MTGuard.cpp:139:28: error: reference to 'mutex' is ambiguous
      std::lock_guard<std::mutex> lock(*FilterMutex);
                           ^
../avs_core/core/MTGuard.h:12:9: note: candidate found by name lookup is 'std::mutex'
  class mutex;
        ^
/usr/include/c++/v1/__mutex_base:40:78: note: candidate found by name lookup is 'std::__1::mutex'
class _LIBCPP_TYPE_VIS _LIBCPP_THREAD_SAFETY_ANNOTATION(capability("mutex")) mutex
                                                                             ^
../avs_core/core/MTGuard.cpp:184:28: error: reference to 'mutex' is ambiguous
      std::lock_guard<std::mutex> lock(*FilterMutex);
                           ^
../avs_core/core/MTGuard.h:12:9: note: candidate found by name lookup is 'std::mutex'
  class mutex;
        ^
/usr/include/c++/v1/__mutex_base:40:78: note: candidate found by name lookup is 'std::__1::mutex'
class _LIBCPP_TYPE_VIS _LIBCPP_THREAD_SAFETY_ANNOTATION(capability("mutex")) mutex
                                                                             ^
/usr/include/c++/v1/__mutex_base:104:26: error: member access into incomplete type 'std::__1::lock_guard<std::mutex>::mutex_type' (aka 'std::mutex')
        : __m_(__m) {__m_.lock();}
                         ^
../avs_core/core/MTGuard.cpp:139:35: note: in instantiation of member function 'std::__1::lock_guard<std::mutex>::lock_guard' requested here
      std::lock_guard<std::mutex> lock(*FilterMutex);
                                  ^
../avs_core/core/MTGuard.h:12:9: note: forward declaration of 'std::mutex'
  class mutex;
        ^
In file included from ../avs_core/core/MTGuard.cpp:41:
In file included from /usr/include/c++/v1/mutex:191:
/usr/include/c++/v1/__mutex_base:109:79: error: member access into incomplete type 'std::__1::lock_guard<std::mutex>::mutex_type' (aka 'std::mutex')
    ~lock_guard() _LIBCPP_THREAD_SAFETY_ANNOTATION(release_capability()) {__m_.unlock();}
                                                                              ^
../avs_core/core/MTGuard.cpp:139:35: note: in instantiation of member function 'std::__1::lock_guard<std::mutex>::~lock_guard' requested here
      std::lock_guard<std::mutex> lock(*FilterMutex);
                                  ^
../avs_core/core/MTGuard.h:12:9: note: forward declaration of 'std::mutex'
  class mutex;
        ^
1 warning and 7 errors generated.
[3/6] Building CXX object avs_core/CMa...AvsCore.dir/filters/text-overlay.cpp.o
FAILED: avs_core/CMakeFiles/AvsCore.dir/filters/text-overlay.cpp.o 
/usr/local/bin/clang++  -DAvsCore_EXPORTS -DBUILDING_AVSCORE -I../avs_core/include -Iavs_core -msse2  -fPIC   -std=c++17 -MD -MT avs_core/CMakeFiles/AvsCore.dir/filters/text-overlay.cpp.o -MF avs_core/CMakeFiles/AvsCore.dir/filters/text-overlay.cpp.o.d -o avs_core/CMakeFiles/AvsCore.dir/filters/text-overlay.cpp.o -c ../avs_core/filters/text-overlay.cpp
In file included from ../avs_core/filters/text-overlay.cpp:35:
../avs_core/filters/text-overlay.h:103:25: warning: 'GetFrame' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
  PVideoFrame __stdcall GetFrame(int n, IScriptEnvironment* env);
                        ^
../avs_core/include/avisynth.h:1178:25: note: overridden virtual function is here
  PVideoFrame __stdcall GetFrame(int n, IScriptEnvironment* env) { return child->GetFrame(n, env); }
                        ^
In file included from ../avs_core/filters/text-overlay.cpp:35:
../avs_core/filters/text-overlay.h:133:25: warning: 'GetFrame' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
  PVideoFrame __stdcall GetFrame(int n, IScriptEnvironment* env);
                        ^
../avs_core/include/avisynth.h:1178:25: note: overridden virtual function is here
  PVideoFrame __stdcall GetFrame(int n, IScriptEnvironment* env) { return child->GetFrame(n, env); }
                        ^
In file included from ../avs_core/filters/text-overlay.cpp:35:
../avs_core/filters/text-overlay.h:202:25: warning: 'GetFrame' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
  PVideoFrame __stdcall GetFrame(int n, IScriptEnvironment* env);
                        ^
../avs_core/include/avisynth.h:1178:25: note: overridden virtual function is here
  PVideoFrame __stdcall GetFrame(int n, IScriptEnvironment* env) { return child->GetFrame(n, env); }
                        ^
In file included from ../avs_core/filters/text-overlay.cpp:35:
../avs_core/filters/text-overlay.h:235:25: warning: 'GetFrame' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
  PVideoFrame __stdcall GetFrame(int n, IScriptEnvironment* env);
                        ^
../avs_core/include/avisynth.h:1178:25: note: overridden virtual function is here
  PVideoFrame __stdcall GetFrame(int n, IScriptEnvironment* env) { return child->GetFrame(n, env); }
                        ^
In file included from ../avs_core/filters/text-overlay.cpp:35:
../avs_core/filters/text-overlay.h:236:18: warning: 'GetParity' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
  bool __stdcall GetParity(int n);
                 ^
../avs_core/include/avisynth.h:1181:18: note: overridden virtual function is here
  bool __stdcall GetParity(int n) { return child->GetParity(n); }
                 ^
In file included from ../avs_core/filters/text-overlay.cpp:35:
../avs_core/filters/text-overlay.h:269:25: warning: 'GetFrame' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
  PVideoFrame __stdcall GetFrame(int n, IScriptEnvironment* env);
                        ^
../avs_core/include/avisynth.h:1178:25: note: overridden virtual function is here
  PVideoFrame __stdcall GetFrame(int n, IScriptEnvironment* env) { return child->GetFrame(n, env); }
                        ^
In file included from ../avs_core/filters/text-overlay.cpp:36:
../avs_core/filters/../convert/convert.h:77:25: warning: 'GetFrame' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
  PVideoFrame __stdcall GetFrame(int n, IScriptEnvironment* env);
                        ^
../avs_core/include/avisynth.h:1178:25: note: overridden virtual function is here
  PVideoFrame __stdcall GetFrame(int n, IScriptEnvironment* env) { return child->GetFrame(n, env); }
                        ^
In file included from ../avs_core/filters/text-overlay.cpp:36:
../avs_core/filters/../convert/convert.h:99:25: warning: 'GetFrame' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
  PVideoFrame __stdcall GetFrame(int n, IScriptEnvironment* env);
                        ^
../avs_core/include/avisynth.h:1178:25: note: overridden virtual function is here
  PVideoFrame __stdcall GetFrame(int n, IScriptEnvironment* env) { return child->GetFrame(n, env); }
                        ^
In file included from ../avs_core/filters/text-overlay.cpp:45:
In file included from /usr/include/c++/v1/sstream:175:
/usr/include/c++/v1/istream:326:26: error: implicit instantiation of undefined template 'std::__1::ctype<char16_t>'
                if (!__ct.is(__ct.space, *__i))
                         ^
/usr/include/c++/v1/istream:1389:57: note: in instantiation of member function 'std::__1::basic_istream<char16_t, std::__1::char_traits<char16_t> >::sentry::sentry' requested here
        typename basic_istream<_CharT, _Traits>::sentry __sen(__is, true);
                                                        ^
../avs_core/filters/text-overlay.cpp:2958:15: note: in instantiation of function template specialization 'std::__1::getline<char16_t, std::__1::char_traits<char16_t>, std::__1::allocator<char16_t> >' requested here
  while (std::getline(wss, temp, u'\n')) {
              ^
/usr/include/c++/v1/__locale:475:52: note: template is declared here
template <class _CharT> class _LIBCPP_TEMPLATE_VIS ctype;
                                                   ^
/usr/include/c++/v1/__locale:234:54: error: implicit instantiation of undefined template 'std::__1::ctype<char16_t>'
    return static_cast<const _Facet&>(*__l.use_facet(_Facet::id));
                                                     ^
/usr/include/c++/v1/istream:322:41: note: in instantiation of function template specialization 'std::__1::use_facet<std::__1::ctype<char16_t> >' requested here
            const ctype<_CharT>& __ct = use_facet<ctype<_CharT> >(__is.getloc());
                                        ^
/usr/include/c++/v1/istream:1389:57: note: in instantiation of member function 'std::__1::basic_istream<char16_t, std::__1::char_traits<char16_t> >::sentry::sentry' requested here
        typename basic_istream<_CharT, _Traits>::sentry __sen(__is, true);
                                                        ^
../avs_core/filters/text-overlay.cpp:2958:15: note: in instantiation of function template specialization 'std::__1::getline<char16_t, std::__1::char_traits<char16_t>, std::__1::allocator<char16_t> >' requested here
  while (std::getline(wss, temp, u'\n')) {
              ^
/usr/include/c++/v1/__locale:475:52: note: template is declared here
template <class _CharT> class _LIBCPP_TEMPLATE_VIS ctype;
                                                   ^
8 warnings and 2 errors generated.
[4/6] Building CXX object avs_core/CMakeFiles/AvsCore.dir/core/info.cpp.o
FAILED: avs_core/CMakeFiles/AvsCore.dir/core/info.cpp.o 
/usr/local/bin/clang++  -DAvsCore_EXPORTS -DBUILDING_AVSCORE -I../avs_core/include -Iavs_core -msse2  -fPIC   -std=c++17 -MD -MT avs_core/CMakeFiles/AvsCore.dir/core/info.cpp.o -MF avs_core/CMakeFiles/AvsCore.dir/core/info.cpp.o.d -o avs_core/CMakeFiles/AvsCore.dir/core/info.cpp.o -c ../avs_core/core/info.cpp
In file included from ../avs_core/core/info.cpp:22:
In file included from ../avs_core/core/info.h:28:
In file included from /usr/include/c++/v1/sstream:175:
/usr/include/c++/v1/istream:326:26: error: implicit instantiation of undefined template 'std::__1::ctype<char16_t>'
                if (!__ct.is(__ct.space, *__i))
                         ^
/usr/include/c++/v1/istream:1389:57: note: in instantiation of member function 'std::__1::basic_istream<char16_t, std::__1::char_traits<char16_t> >::sentry::sentry' requested here
        typename basic_istream<_CharT, _Traits>::sentry __sen(__is, true);
                                                        ^
../avs_core/core/info.cpp:7265:15: note: in instantiation of function template specialization 'std::__1::getline<char16_t, std::__1::char_traits<char16_t>, std::__1::allocator<char16_t> >' requested here
  while (std::getline(wss, temp, u'\n'))
              ^
/usr/include/c++/v1/__locale:475:52: note: template is declared here
template <class _CharT> class _LIBCPP_TEMPLATE_VIS ctype;
                                                   ^
/usr/include/c++/v1/__locale:234:54: error: implicit instantiation of undefined template 'std::__1::ctype<char16_t>'
    return static_cast<const _Facet&>(*__l.use_facet(_Facet::id));
                                                     ^
/usr/include/c++/v1/istream:322:41: note: in instantiation of function template specialization 'std::__1::use_facet<std::__1::ctype<char16_t> >' requested here
            const ctype<_CharT>& __ct = use_facet<ctype<_CharT> >(__is.getloc());
                                        ^
/usr/include/c++/v1/istream:1389:57: note: in instantiation of member function 'std::__1::basic_istream<char16_t, std::__1::char_traits<char16_t> >::sentry::sentry' requested here
        typename basic_istream<_CharT, _Traits>::sentry __sen(__is, true);
                                                        ^
../avs_core/core/info.cpp:7265:15: note: in instantiation of function template specialization 'std::__1::getline<char16_t, std::__1::char_traits<char16_t>, std::__1::allocator<char16_t> >' requested here
  while (std::getline(wss, temp, u'\n'))
              ^
/usr/include/c++/v1/__locale:475:52: note: template is declared here
template <class _CharT> class _LIBCPP_TEMPLATE_VIS ctype;
                                                   ^
2 errors generated.
[5/6] Building CXX object avs_core/CMakeFiles/AvsCore.dir/core/avisynth.cpp.o
FAILED: avs_core/CMakeFiles/AvsCore.dir/core/avisynth.cpp.o 
/usr/local/bin/clang++  -DAvsCore_EXPORTS -DBUILDING_AVSCORE -I../avs_core/include -Iavs_core -msse2  -fPIC   -std=c++17 -MD -MT avs_core/CMakeFiles/AvsCore.dir/core/avisynth.cpp.o -MF avs_core/CMakeFiles/AvsCore.dir/core/avisynth.cpp.o.d -o avs_core/CMakeFiles/AvsCore.dir/core/avisynth.cpp.o -c ../avs_core/core/avisynth.cpp
In file included from ../avs_core/core/avisynth.cpp:74:
../avs_core/core/MTGuard.h:22:8: error: reference to 'mutex' is ambiguous
  std::mutex *FilterMutex;
       ^
/usr/include/c++/v1/__mutex_base:40:78: note: candidate found by name lookup is 'std::__1::mutex'
class _LIBCPP_TYPE_VIS _LIBCPP_THREAD_SAFETY_ANNOTATION(capability("mutex")) mutex
                                                                             ^
../avs_core/core/MTGuard.h:12:9: note: candidate found by name lookup is 'std::mutex'
  class mutex;
        ^
../avs_core/core/MTGuard.h:33:8: error: reference to 'mutex' is ambiguous
  std::mutex* GetMutex() const;
       ^
/usr/include/c++/v1/__mutex_base:40:78: note: candidate found by name lookup is 'std::__1::mutex'
class _LIBCPP_TYPE_VIS _LIBCPP_THREAD_SAFETY_ANNOTATION(capability("mutex")) mutex
                                                                             ^
../avs_core/core/MTGuard.h:12:9: note: candidate found by name lookup is 'std::mutex'
  class mutex;
        ^
In file included from ../avs_core/core/avisynth.cpp:730:
../avs_core/core/vartable.h:75:16: error: reference to 'mutex' is ambiguous
  mutable std::mutex var_mutex;
               ^
/usr/include/c++/v1/__mutex_base:40:78: note: candidate found by name lookup is 'std::__1::mutex'
class _LIBCPP_TYPE_VIS _LIBCPP_THREAD_SAFETY_ANNOTATION(capability("mutex")) mutex
                                                                             ^
../avs_core/core/MTGuard.h:12:9: note: candidate found by name lookup is 'std::mutex'
  class mutex;
        ^
In file included from ../avs_core/core/avisynth.cpp:730:
../avs_core/core/vartable.h:95:26: error: reference to 'mutex' is ambiguous
    std::lock_guard<std::mutex> lock(var_mutex); // avoid concurrency for global variables
                         ^
/usr/include/c++/v1/__mutex_base:40:78: note: candidate found by name lookup is 'std::__1::mutex'
class _LIBCPP_TYPE_VIS _LIBCPP_THREAD_SAFETY_ANNOTATION(capability("mutex")) mutex
                                                                             ^
../avs_core/core/MTGuard.h:12:9: note: candidate found by name lookup is 'std::mutex'
  class mutex;
        ^
In file included from ../avs_core/core/avisynth.cpp:730:
../avs_core/core/vartable.h:111:26: error: reference to 'mutex' is ambiguous
    std::lock_guard<std::mutex> lock(var_mutex); // avoid concurrency for global variables
                         ^
/usr/include/c++/v1/__mutex_base:40:78: note: candidate found by name lookup is 'std::__1::mutex'
class _LIBCPP_TYPE_VIS _LIBCPP_THREAD_SAFETY_ANNOTATION(capability("mutex")) mutex
                                                                             ^
../avs_core/core/MTGuard.h:12:9: note: candidate found by name lookup is 'std::mutex'
  class mutex;
        ^
../avs_core/core/avisynth.cpp:821:8: error: reference to 'mutex' is ambiguous
  std::mutex string_mutex;
       ^
/usr/include/c++/v1/__mutex_base:40:78: note: candidate found by name lookup is 'std::__1::mutex'
class _LIBCPP_TYPE_VIS _LIBCPP_THREAD_SAFETY_ANNOTATION(capability("mutex")) mutex
                                                                             ^
../avs_core/core/MTGuard.h:12:9: note: candidate found by name lookup is 'std::mutex'
  class mutex;
        ^
../avs_core/core/avisynth.cpp:1352:26: error: reference to 'mutex' is ambiguous
    std::lock_guard<std::mutex> lock(string_mutex);
                         ^
/usr/include/c++/v1/__mutex_base:40:78: note: candidate found by name lookup is 'std::__1::mutex'
class _LIBCPP_TYPE_VIS _LIBCPP_THREAD_SAFETY_ANNOTATION(capability("mutex")) mutex
                                                                             ^
../avs_core/core/MTGuard.h:12:9: note: candidate found by name lookup is 'std::mutex'
  class mutex;
        ^
../avs_core/core/avisynth.cpp:3215:24: error: reference to 'mutex' is ambiguous
  std::lock_guard<std::mutex> lock(string_mutex);
                       ^
/usr/include/c++/v1/__mutex_base:40:78: note: candidate found by name lookup is 'std::__1::mutex'
class _LIBCPP_TYPE_VIS _LIBCPP_THREAD_SAFETY_ANNOTATION(capability("mutex")) mutex
                                                                             ^
../avs_core/core/MTGuard.h:12:9: note: candidate found by name lookup is 'std::mutex'
  class mutex;
        ^
../avs_core/core/avisynth.cpp:3223:26: error: reference to 'mutex' is ambiguous
    std::lock_guard<std::mutex> lock(string_mutex);
                         ^
/usr/include/c++/v1/__mutex_base:40:78: note: candidate found by name lookup is 'std::__1::mutex'
class _LIBCPP_TYPE_VIS _LIBCPP_THREAD_SAFETY_ANNOTATION(capability("mutex")) mutex
                                                                             ^
../avs_core/core/MTGuard.h:12:9: note: candidate found by name lookup is 'std::mutex'
  class mutex;
        ^
9 errors generated.
ninja: build stopped: subcommand failed.
[qyot27@bsdhost ~/AviSynthPlus/avisynth-build]$ 
```

macOS shows extremely similar/identical build failures with either the default Clang or with LLVM-9/Clang-9, regarding MTGuard.

**Comment by pinterf** [comment]
In MtGuard.h let's try the following:
insert
`#include <mutex>`

and comment out
```
/*
namespace std
{
  class mutex;
}
*/

Windows build (both MSVC and ClangCL)was O.K. with it. Probably it was needed for an earlier MS compiler in 2013-14?
```

**Comment by qyot27** [comment]
The mutex changes appear to also work fine on macOS/BSD as well.  Which leaves the remaining two issues:

A) Errors from inside info.h/.cpp and text-overlay.cpp regarding `<sstream>` and the `implicit initialization of undefined template 'std::__1::ctype<char16_t>'`.  According to the results I found with Google, this might mean just a missing header, but all the ones suggested to fix the issue didn't.  It also doesn't really explain why Clang-9 on Ubuntu is cool with it.

B) Warnings about overrides that didn't have the override keyword set.  Thankfully, these aren't a fatal error.

**Comment by pinterf** [comment]
I'll figure out that sstream and char16 thing, must be similar like the mutex was, just a bit deeper inside. I noticed those warnings as well, they are not on priority, when I'm fed of them I usually clean up different kinds of warning families, still at least three big families survive here and there :)

**Comment by pinterf** [comment]
sstream::get_line has an is_space internally. That seems to be not specialized for char16_t. I reverted to wchar_t in general.

**Comment by qyot27** [comment]
Okay, so testing on FreeBSD and macOS again, and FreeBSD is now in the clear: the default Clang 8 can build it, and so can Clang 9.  macOS does not, because `<filesystem>` isn't present in the libc++ that Apple ships by default (except for Catalina/10.15, apparently).  High Sierra/10.13 and Mojave/10.14 may very well have to stick to either building with GCC or if I can, figure out how to really use upstream LLVM for real without the default one overriding it.

**Comment by pinterf** [comment]
Well, I have finished a huge cherrypick-merge (you can see it in my fork atm) that brings up all the smart things that Avisynth Neo project is achieved, while maintaning our big changes from the last years, posix, the good old array concept, and keeping ScriptClip compatibility as well. 

You can check those in readme_history.txt .

Since I wanted to keep the credits and history I cherry picked step-by-step, one feature at a time, one fix group at a time, until it became obvious that I could not pick just one fix or feature because the whole core was rethinked and made smarter. The were regressions after the merge, which I had fixed (hopefully). This procedure surely introduced a lot of code cosmetical glitches, I'd say, we should wait a month until the code clean. On one hand tabs vs space CR vs CR LF made the merge and comparison process extremely difficult. On the other hand, there will be more (but not so big) changes, we can do the source beautifying later.

There are some changes already in our upstream master, my repo is a bit behind of it. qyot27, what do you recommend? Can I start a pull request (sor far merge could be done automatically by github engine), now I'm not sure that itcan do it w/o problems.).

**Comment by qyot27** [comment]
What I want to do is make sure we have a minor bugfix release, 3.5.1, that includes all the cleaning-up and streamlining of the POSIX side from the last few weeks, since most of the kinks have been hammered out as far as those go and those not wanting to use the git HEAD will be stuck with the 3.5.0 release tarball, which doesn't include those quality-of-life changes.

The only critical one remaining that's preventing 3.5.1, IMO, is #133.  I have no clue how to have the core read the file and use the value(s), though.  I also need to tidy up some of the instructions in posix.rst since `-fpermissive` is no longer needed, amongst other things.

Then I think we could go forward with looking at what's necessary to merge in the Neo changes.  If you want to open a pull request now, that'd be fine, though.

**Comment by qyot27** [comment]
After fighting with it more last night, trying to cobble together reading the config file and getting the values out, I came to the conclusion that it's probably not worth going to all that extra trouble.

Instead, I simply hardcoded two values that the config file would have handled otherwise: the `avisynth` subdirectory in the regular /lib install location (turning avisynth.conf.in into avisynth_conf.h.in and no longer installing it), and `$HOME/.avisynth`.  Because in reality, those are the two most vital locations to handle, and if a user wants to automatically autoload more directories, it should still be possible for a regular .avsi to handle it with normal AddAutoloadDir use, placed in one of those two autoload directories.

I've not tested it on anything except Linux yet, and #143 needs to be broken up into topical commits (I've more or less come to the conclusion that the SOVERSION stuff will just have to look weird, as my attempts to make it handled more gracefully caused loading to fail).
